### PR TITLE
only allocate trans_ for globalTrans_

### PR DESCRIPTION
**Pull request by GitPaean**
others are not needed for globalTrans_ for now. It will help to flatten the peak memory usage for rank 0 during a parallel running.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
With this PR, for the 1 million case SPE11C, with 14 processes, rank 0, it reduces the peak memory usage from 2.26 GB to 2 GB. And the difference will scale following the number of the faces globally. 
Master:
image

PR:
image

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/flow/Transmissibility.hpp [hunk 1]
```diff
@@ -147,7 +152,7 @@ class Transmissibility {
      *   numerical aquifers.  Default value: \c false, meaning do not apply
      *   regional multipliers to explicit NNCs.
      */
-    void update(bool global, const std::function<unsigned int(unsigned int)>& map = {}, bool applyNncMultRegT = false);
+    void update(bool global, bool onlyTrnas = false, const std::function<unsigned int(unsigned int)>& map = {}, bool applyNncMultRegT = false);
```
Comment (blattms) [inline opm/simulators/flow/Transmissibility.hpp:155]: small typo, that does not really matter: onlyTrans

**Review by blattms — APPROVED** [review]
Looks good.

If I understand this correctly, then this relevant before loadbalancing. With this PR we will only create the gloabl transmissibility arrays needed to feed to the loadbalancer.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
I do see this will introduced a `incomplete` `TransmissbilityType`, which might pose problem for certain usage.  Maybe we should introduce the `onlyTrans` as member to the `TransmissbilityType` or something similar, so we can indicate what is offered from the object.

**Comment by GitPaean** [comment]
Adding some comments to emphasize that only trans_ in the globalTrans_, so hopefully people can follow the issue.

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/flow/CpGridVanguard.hpp [hunk 2]
```diff
@@ -306,6 +306,10 @@ class CpGridVanguard : public FlowBaseVanguard<TypeTag>
         this->doFilterConnections_(this->schedule());
     }
 
+    // \Note: this globalTrans_ is used for domain decompostion and INIT file output.
```
Comment (bska) [inline opm/simulators/flow/CpGridVanguard.hpp:309]: Tiny misprint: "decompostion" ➡️ "decomposition".

**Review by bska — COMMENTED** [review]
This is okay, but from a readability point of view I think I'd prefer that we use named enumerators instead of naked `bool`s which cannot be fully understood in isolation.  Although borderline now with just two `bool` parameters we might consider something like
```C++
struct TransUpdate {
    enum class Ranks { Zero, All, };
    enum class Quantities { Trans, All, };

    Ranks ranks;
    Quantities quantities;
};
```
and then one'd call `update()` as
```C++
update({ TransUpdate::Ranks::All, TransUpdate::Quantities::Trans }, /* ... */);
```
which, at least to me, is easier to read/understand than
```C++
update(true, false, /* ... */)
```

**Comment by GitPaean** [comment]
> This is okay, but from a readability point of view I think I'd prefer that we use named enumerators instead of naked `bool`s which cannot be fully understood in isolation. Although borderline now with just two `bool` parameters we might consider something like
> 
> ```c++
> struct TransUpdate {
>     enum class Ranks { Zero, All, };
>     enum class Quantities { Trans, All, };
> 
>     Ranks ranks;
>     Quantities quantities;
> };
> ```
> 
> and then one'd call `update()` as
> 
> ```c++
> update({ TransUpdate::Ranks::All, TransUpdate::Quantities::Trans }, /* ... */);
> ```
> 
> which, at least to me, is easier to read/understand than
> 
> ```c++
> update(true, false, /* ... */)
> ```

this is a good idea and will incorporate.

**Comment by GitPaean** [comment]
At the end I decided to only use `enum class` for onlyTrans parameter. Did not feel it is improving for the `global` parameter. 

And also, there is default values for onlyTrans while not for global. It affects the usage a little bit.

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  This looks good to me now and I'll merge into master.

**Pull request merged**